packages/ts/src/components/crosshair/index.ts

```diff
@@ -50,44 +53,47 @@ export class Crosshair<Datum> extends XYComponentCore<Datum, CrosshairConfigInte
   public set accessors (accessors: CrosshairAccessors<Datum>) { this._accessors = accessors }
   public get accessors (): CrosshairAccessors<Datum> {
     const { config } = this
-
-    const hasConfig = !!(config.x || config.y || config.yStacked)
-    const x = hasConfig ? config.x : this._accessors.x
-    const yAcc = hasConfig ? config.y : this._accessors.y
+    // If x or y are explicitly set in config, use those; otherwise use container accessors
+    const x = config.x || this._accessors.x
+    const yAcc = config.y || this._accessors.y
     const y = yAcc ? (isArray(yAcc) ? yAcc : [yAcc]) : undefined
-    const yStacked = hasConfig ? config.yStacked : this._accessors.yStacked
+    const yStacked = config.yStacked || this._accessors.yStacked
     const baseline = config.baseline ?? this._accessors.baseline
 
     return { x, y, yStacked, baseline }
   }
 
   constructor (config?: CrosshairConfigInterface<Datum>) {
     super()
-    if (config) this.setConfig(config)
+    if (config) this.config = { ...this.config, ...config }
 
     this.g.style('opacity', this.show ? 1 : 0)
     this.line = this.g.append('line')
       .attr('class', s.line)
   }
 
   setContainer (containerSvg: Selection<SVGSVGElement, unknown, SVGSVGElement, unknown>): void {
-    // Set up mousemove event for Crosshair
     this.container = containerSvg
     this.container.on('mousemove.crosshair', this._onMouseMove.bind(this))
     this.container.on('mouseout.crosshair', this._onMouseOut.bind(this))
+    this._render()
   }
 
   _render (customDuration?: number): void {
     const { config } = this
-    if (config.snapToData && !this.datum) return
     const duration = isNumber(customDuration) ? customDuration : config.duration
+    const { isActive, enableSync } = this._getSyncState()
```

Comment:
I'm not sure it's worth introducing the whole concept of `sync` into the component because, logically, all the syncing is happening outside of it. What we do on the Crosshair component is simply enforce it to be displayed at a specific position.